File: e2e/playwright-python/support/mocks/wiremock_client.py
import requests
import config


class WireMockClient:
    def __init__(self):
        self._base_url = config.wiremock_url

    def clear_mocks(self) -> "WireMockClient":
        requests.delete(f"{self._base_url}/__admin/mappings").raise_for_status()
        return self

    def stub_task_validation(self, is_valid: bool) -> "WireMockClient":
        requests.post(
            f"{self._base_url}/__admin/mappings",
            json={
                "request": {
                    "method": "GET",
                    "urlPattern": "/api/tasks/.*/validation",
                },
                "response": {
                    "status": 200,
                    "jsonBody": {"valid": is_valid},
                },
            },
        ).raise_for_status()
        return self

File: e2e/capybara-ruby/support/mocks/wire_mock_client.rb
# frozen_string_literal: true

require 'faraday'

require_relative '../test_config'

class WireMockClient
  def initialize
    @base_url = TestConfig.wiremock_url
    @connection = Faraday.new(url: @base_url) do |faraday|
      faraday.request :json
      faraday.response :json
      faraday.adapter Faraday.default_adapter
    end
  end

  def clear_mocks
    @connection.delete('__admin/mappings')
    self
  end

  def stub_task_validation(is_valid)
    @connection.post('__admin/mappings', {
      request: {
        method: 'GET',
        urlPattern: '/api/tasks/.*/validation'
      },
      response: {
        status: 200,
        jsonBody: { valid: is_valid }
      }
    })
    self
  end
end
